pkg/api/uploads.go

```diff
@@ -1,16 +1,33 @@
 package api
 
+import "time"
+
 type AddUploadsRequest struct {
 	Uploads   []Upload   `json:"uploads"`   // List of unfinished uploads
 	Artifacts []Artifact `json:"artifacts"` // List of created artifacts
 }
 
 type Upload struct {
-	Href   string `json:"href"`   // HREF to the unfinished upload
+	Uuid   string `json:"uuid"`   // Upload UUID, use with public API
+	Href   string `json:"href"`   // HREF to the unfinished upload, use with internal API
 	Sha256 string `json:"sha256"` // SHA256 sum of the uploaded file
 }
 
 type Artifact struct {
 	Href   string // HREF to the  completed artifact
 	Sha256 string // SHA256 sum of the completed artifact
 }
+
+type PublicUploadChunkRequest struct {
+	UploadUuid string `param:"upload_uuid"` // Upload UUID
+	File       string `form:"file"`         // A chunk of the uploaded file
+	Sha256     string `form:"sha256"`       // SHA-256 checksum of the chunk
+}
```

Comment:
Since this API is meant to be user-friendly, I would rename this "UploadChunkRequest", and rename the internal pulp one to "PulpUploadChunkRequest". I could see the term "public upload" causing confusion.